File: freecodecamp/daily-code-challenge/2026-01-30.js
function findPawnMoves(position) {
  const col = position[0];
  const row = parseInt(position[1], 10);
  const moves = [];

  // one square forward
  if (row < 8) {
    moves.push(`${col}${row + 1}`);
  }

  // two squares forward if on starting rank
  if (row === 2) {
    moves.push(`${col}${row + 2}`);
  }

  return moves.sort();
}

File: freecodecamp/daily-code-challenge/2026-01-30.py
def find_pawn_moves(position):
    col = position[0]
    row = int(position[1])
    moves = []

    # one square forward
    if row < 8:
        moves.append(f"{col}{row + 1}")

    # two squares forward if on starting rank
    if row == 2:
        moves.append(f"{col}{row + 2}")

    return sorted(moves)
